ui/src/text_annotator.tsx

```diff
@@ -227,22 +260,47 @@ export
             <span style={removeIconStyle as React.CSSProperties} className={css.iconUnderlay}></span>
           </mark>
         )
-      }
+      },
+      handleMouseDown = (startElProps: TokenMouseEventProps) => () => {
+        startElPropsRef.current = startElProps
+      },
+      handleMouseUp = (endElProps: TokenMouseEventProps) => () => {
+        annotate(endElProps)
+      },
+      handleMouseLeave = (endElProps: TokenMouseEventProps) => (ev: React.MouseEvent<HTMLSpanElement>) => {
+        if (startElPropsRef.current !== undefined && (ev.relatedTarget as any)?.nodeName !== 'P') annotate(endElProps) // nodeName prop is not typed yet
+      },
+      text = tokens.map((token, idx) => {
+        return <Token
+          key={idx}
+          idx={idx}
+          tokenProps={token}
+          handleMouseDown={handleMouseDown}
+          handleMouseUp={handleMouseUp}
+          handleMouseLeave={handleMouseLeave}
+          getMark={getMark}
+          activeColor={activeTag ? tagColorMap.get(activeTag)! : undefined}
+        />
+      }, [] as React.ReactElement[])
```

Comment:
nit

```suggestion
      text = tokens.map((token, idx) => (<Token
          key={idx}
          idx={idx}
          tokenProps={token}
          handleMouseDown={handleMouseDown}
          handleMouseUp={handleMouseUp}
          handleMouseLeave={handleMouseLeave}
          getMark={getMark}
          activeColor={activeTag ? tagColorMap.get(activeTag) : undefined}
        />
      ))
```
Can also be part of the JSX - probably better.